core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -1261,6 +1455,11 @@ public Builder withDatafile(String datafile) {
             return this;
         }
 
+        public Builder withDefaultDecideOptions(List<OptimizelyDecideOption> options) {
+            this.defaultDecideOptions = Collections.unmodifiableList(options);
```

Comment:
I think we can revise this and instead of converting options to unmodifiableList here we can just assign it and in build method we can check if options is not null, then change it to unmodifiableList because it will cause null pointer exception.